force-app/main/default/classes/StorageChargeCheck.cls

```diff
@@ -0,0 +1,62 @@
+public with sharing class StorageChargeCheck implements Database.Batchable<sObject>
+{
+    public Database.QueryLocator start(Database.BatchableContext bc)
+    {
+        //Get all purchased cars
+        String query= 'Select id, Purchase_Date__c From Car__c Where Purchase_Date__c != null';
+        return Database.getQueryLocator(query);
+    }
+
+    public void execute(Database.BatchableContext bc, List<sObject> scope)
+    {
+        Integer count = 0;
+        //Iterate purchased cars
+        for (Car__c c : (List<Car__c>)scope)
+        {
+            //Get purchased date
+            Date purchasedDate = date.newinstance(c.Purchase_Date__c.year(), c.Purchase_Date__c.month(), c.Purchase_Date__c.day());
```

Comment:
why not to use c.Purchase_Date__c.date() it will return date

Reply:
I didn't know about it😆